### PR TITLE
.github: add timeout for individual tests + bump dependencies

### PR DESCRIPTION
**Pull request by evgeniiz321**

### CODE REVIEW

**Comment by evgeniiz321** [comment]
@roman-khimov can you please check repo settings once again? Maybe we need to remove and add once again the rule for run_system_tests job.

**Comment by roman-khimov** [comment]
Deleted/added back. Let's see how it goes.

**Comment by evgeniiz321** [comment]
@roman-khimov I tried setting rules on my demo repo and I can select individual matrix jobs there:

image

can you do the same here?

**Comment by roman-khimov** [comment]
Of course I can. But it allows to select parent entity as well which doesn't work. Weird.

File: .github/workflows/system-tests.yml [hunk 1]
```diff
@@ -112,20 +112,20 @@ jobs:
         timeout-minutes: 500
         steps:
           - name: Set up Go
-            uses: actions/setup-go@v5
+            uses: actions/setup-go@v5.1.0
```
Comment (roman-khimov) [inline .github/workflows/system-tests.yml:115]: This shouldn't be needed, v5 tracks the latest v5.X.Y.

File: .github/workflows/system-tests.yml [hunk 2]
```diff
@@ -180,7 +180,7 @@ jobs:
 
           - name: Download neofs-cli
             if: ${{ inputs.neofs_node_tag != 'from_tests' }}
-            uses: dsaltares/fetch-gh-release-asset@1.1.1
+            uses: dsaltares/fetch-gh-release-asset@1.1.2
```
Comment (roman-khimov) [inline .github/workflows/system-tests.yml:183]: Does it have a `v1` of some kind?

File: .github/workflows/system-tests.yml [hunk 3]
```diff
@@ -372,7 +372,7 @@ jobs:
             if: always() && steps.put_report.outcome == 'success'
             env:
               REPORT_NEOFS_URL: ${{ steps.put_report.outputs.REPORT_NEOFS_URL }}index.html
-            uses: Sibz/github-status-action@v1
+            uses: guibranco/github-status-action-v2@v1.1.13
```
Comment (roman-khimov) [inline .github/workflows/system-tests.yml:375]: That's different. Are we sure it's OK to use?

File: .github/workflows/system-tests.yml (hunk 1 above)
Comment (roman-khimov) [reply .github/workflows/system-tests.yml:115]: Same applies to other @vX

File: .github/workflows/system-tests.yml (hunk 3 above)
Comment (evgeniiz321) [reply .github/workflows/system-tests.yml:375]: the old one is deprecated

File: .github/workflows/system-tests.yml (hunk 2 above)
Comment (evgeniiz321) [reply .github/workflows/system-tests.yml:183]: Nope

File: .github/workflows/system-tests.yml (hunk 3 above)
Comment (roman-khimov) [reply .github/workflows/system-tests.yml:375]: Looks like not a lot of choices at the moment.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**